src/modules/workspace/hooks/useGifLoadingRequest.ts

```diff
@@ -1,15 +1,35 @@
 import { useQuery } from '@tanstack/react-query';
 
+import { authCredentials } from '@/modules';
+import { IUserInfos } from '@/modules/auth/services';
+import { useAuthStore } from '@/modules/auth/store/useAuthStore';
+
 export enum GifLoadingRequestQueryKey {
   ANIMATION_LOADING = 'animation-loading',
 }
 
-const useGitLoadingRequest = () => {
+const useGitLoadingRequest = (logout?: () => void, userInfos?: IUserInfos) => {
+  const { isTokenExpired, setIsTokenExpired } = useAuthStore();
+  const { address, token } = authCredentials();
+
   const { isLoading, isFetching, refetch } = useQuery({
     queryKey: [GifLoadingRequestQueryKey.ANIMATION_LOADING],
     queryFn: () =>
       new Promise((resolve) => {
+        if (
+          !token &&
+          !address &&
+          !userInfos?.address &&
+          !isLoading &&
+          !isFetching
+        ) {
```

Comment:
I would like to explain the reason for this `if` in the line 19.

The logic is: when the query/promise STARTS, we do a first validation: We verify the token and adress from cookies but if we only use those, when the user log in, before he can see the home page, he's thrown to sigin page again. 
So, the `!userInfos?.address`  is to check if he just logged in, and the `isLoading` and the `isFetching` are to wait the first gif animation, because whenever an log in/goes to home page, the gif is shown. So we need to wait this first gif animation to do the validation.